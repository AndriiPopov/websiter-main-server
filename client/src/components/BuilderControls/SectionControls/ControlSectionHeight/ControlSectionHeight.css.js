export const SectionHeight = {
    userSelect: 'none',
    position: 'absolute',
    bottom: '-6px',
    height: '13px',
    background: '#eee',
    border: '1px solid #ccc',
    boxSizing: 'border-box',
    cursor: 'row-resize',
    fontSize: '11px',
    color: '#555',
    fontFamily: 'arial',
    padding: '0px 3px',
    zIndex: '1000',
    left: '0px',
    right: '0px',
    margin: '0px 50px',
    width: '45px',
    lineHeight: '10px'
};