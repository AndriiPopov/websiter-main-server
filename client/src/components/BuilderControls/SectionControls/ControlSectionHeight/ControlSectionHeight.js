import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import * as classes from './ControlSectionHeight.css.js';
import * as actions from '../../../../store/actions/index';
import { dragStart } from '../../../../utils/dragFunction';

export const ControlSectionHeight = props => {
    const handleDragMouseDown = (e) => {
        const startElY = props.section.height;

        dragStart(e, (difX, difY) => {
            let value = startElY + difY;
            if (value < 0) {
                value = 0;
            }
            props.changed(value, 'height');
        }, () => {});
    };

    return !props.previewIsOn ?
        <div style={classes.SectionHeight}
            onMouseDown={handleDragMouseDown}>
            Height
        </div>
        : null;
};

ControlSectionHeight.propTypes = {
    section: PropTypes.shape({
        type: PropTypes.oneOf(['section']).isRequired,
        height: PropTypes.number.isRequired,
        children: PropTypes.arrayOf(PropTypes.string).isRequired,
        styles: PropTypes.arrayOf(PropTypes.string).isRequired
    }).isRequired,
    id: PropTypes.string.isRequired
};

const mapStateToProps = (state, props) => {
    return {
        section: state.builder.present[props.id],
        previewIsOn: state.builder.present.preview        
    };
};

const mapDispatchToProps = dispatch => {
    return {
        changed: (newValue, valueKey) => dispatch(actions.changeElementPropertyValue(newValue, valueKey, true))
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(ControlSectionHeight);