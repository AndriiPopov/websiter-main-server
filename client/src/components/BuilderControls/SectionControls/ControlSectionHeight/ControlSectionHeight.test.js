import React from 'react';
import renderer from 'react-test-renderer';

import { ControlSectionHeight } from './ControlSectionHeight';

describe('BoxControls', () => {
    it('should be defined', () => {
        expect(ControlSectionHeight).toBeDefined();
    });

    it('it should render 1 div', () => {
        const tree = renderer.create(
            <ControlSectionHeight />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});