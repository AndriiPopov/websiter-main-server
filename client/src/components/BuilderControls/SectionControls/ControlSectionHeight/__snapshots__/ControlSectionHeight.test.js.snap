// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoxControls it should render 1 div 1`] = `
<div
  onMouseDown={[Function]}
  style={
    Object {
      "background": "#eee",
      "border": "1px solid #ccc",
      "bottom": "-6px",
      "boxSizing": "border-box",
      "color": "#555",
      "cursor": "row-resize",
      "fontFamily": "arial",
      "fontSize": "11px",
      "height": "13px",
      "left": "0px",
      "lineHeight": "10px",
      "margin": "0px 50px",
      "padding": "0px 3px",
      "position": "absolute",
      "right": "0px",
      "userSelect": "none",
      "width": "45px",
      "zIndex": "1000",
    }
  }
>
  Height
</div>
`;
